packages/html/src/events/pointer.rs

```diff
@@ -1,33 +1,74 @@
+use std::fmt::Debug;
+
 use dioxus_core::Event;
 
+use crate::point_interaction::{PointData, PointInteraction};
+
+/// A synthetic event that wraps a web-style [`PointerEvent`](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent)
 pub type PointerEvent = Event<PointerData>;
+
+/// Data associated with a pointer event, aside from the data shared with mouse events
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, PartialEq)]
+pub struct PointerEventData {
```

Comment:
Usimg `PointerEventData`, `PointerEvent`, and `PointerData` inside an `Event<>` could get confusing quickly. Instead of having this as a seperate type, can we just allow users to pass in all of the parameters in the new function, or even better expose `.with_pointer_id`, `.with_width`, etc. builder methods on the event?

If not, maybe we could call this `PointerOnlyData`, or `PointerMetadata`? Those are not a lot better, but could be more clear because they are not related to an event

Reply:
I added the PointerEventData struct because I was getting a clippy warning about too many parameters on the ctor. I like the idea of a builder, let me try that

Reply:
I ended up just suppressing the too_many_arguments clippy rule as it seems there's no real reason for it other than style